src/main/resources/data-repository-openapi.yaml

```diff
@@ -1872,16 +1872,49 @@ definitions:
     properties:
       name:
         $ref: '#/definitions/ObjectNameProperty'
-
       columns:
         type: array
         items:
           $ref: '#/definitions/ColumnModel'
-
       primaryKey:
         type: array
         items:
           $ref: '#/definitions/ObjectNameProperty'
+      partitionStrategy:
+        $ref: '#/definitions/PartitionStrategy'
+      partitionColumn:
```

Comment:
Instead of putting partitionColumn here, I suggest breaking out separate structures for each of the strategy types.
Then you make it impossible to express things like: supplying a partition column when it isn't allowed for that strategy.

See how Rori did that for the snapshot interface. I think it is much cleaner. Also, in OpenAPI V3, we can restrict so only one choice can be valid. With V2, we have to validate that the other choices are null.